pysmt/optimization/goal.py

```diff
@@ -125,15 +129,31 @@ def term(self):
 
 class MinMaxGoal(MinimizationGoal):
 
-    def __init__(self, terms):
-        MinimizationGoal.__init__(self, get_env().formula_manager.Max(terms))
+    def __init__(self, terms, sign = True):
+        if len(terms) > 0:
+            if get_env().stc.get_type(terms[0]).is_bv_type():
```

Comment:
It's a bit dirty to use `get_env()` in pysmt classes. The correct solution would be to have an `env` field in the goal classes that is initialized to `get_env()` if no explicit environment is passed. 

This is a bit technical into the pysmt structure. If you have problems, just leave a `#TODO` comment about this here and I'll fix it in the `optimization` branch